section2/number.rb

```diff
@@ -0,0 +1,53 @@
+# Section 2 - Using of Number
+p 10 / 4 # 2
+p 10.0 / 4 # 2.5
+p 10 / 4.0 # 2.5
+p 10 / 4.to_f # 2.5
```

Comment:
explain why.
- int / int = int
- int / float = float
- float / int = float

Reply:
> int / int = int
> int / float = float
> float / int = float

Since Ruby will know the data type like `"hello"` is `String` and `10` is `Int`
The result is depends on the data type like `int / float = float` here, Ruby recognize that there's floating point value object like `10.0`, then the final result is float, since only float can stores floating point values.

For `int / int = int` there isn't any floating point value, thus Ruby returns the result in `int` type.

Reply:
you talk so much but still never answer it. its nothing to do with ruby specific. almost all language behavior the same

Reply:
This follows Usual Arithmetic Conversions which in ASCI C standard, since Ruby is written in C. Which mention that **_if either operand has type float , the other operand is converted to type float ._** which got the result of `float`